SimPEG/electromagnetics/natural_source/simulation_1d.py

```diff
@@ -0,0 +1,433 @@
+        Z_i = \\dfrac{\\alpha_i}{\\sigma_i} \\Bigg [
+        \\dfrac{\\sigma_i Z_{i+1} - \\alpha_i tanh(\\alpha_i h_i)}
+        {\\alpha_i - \\sigma_i Z_{i+1}tanh(\\alpha_i h_i)} \\Bigg ]
+
+    where the complex impedance in the bottom half-space is given by:
+
+    .. math::
+        Z_N = - \\frac{\\alpha_N}{\\sigma_N}
+
+
+    """
+
+    # Must be 1D survey object
+    survey = properties.Instance("a Survey1D survey object", Survey1D, required=True)
+
+    # Finite difference or analytic sensitivities can be computed
+    sensitivity_method = properties.StringChoice(
+        "Choose 1st or 2nd order computations with sensitivity matrix ('1st order', '2nd order')",
+        {
+            "1st order": ["1st_order", "1st-order", "1st", "1", "first"],
+            "2nd order": ["2nd_order", "2nd-order", "2nd", "2", "second"],
+        },
+    )
+
+    # Add layer thickness as invertible property
+    thicknesses, thicknessesMap, thicknessesDeriv = props.Invertible(
+        "thicknesses of the layers"
+    )
+
+    # Storing sensitivity
+    _Jmatrix = None
+    fix_Jmatrix = False
+    storeJ = properties.Bool("store the sensitivity", default=False)
+
+    # Frequency for each datum
+    _frequency_vector = None
+
+    @property
+    def frequency_vector(self):
+        """
+        A vector containing the corresponding frequency for each datum.
+        """
+
+        if getattr(self, "_frequency_vector", None) is None:
+            if self._frequency_vector is None:
+                fvec = []
+                for src in self.survey.source_list:
+                    fvec.append(src.frequency)
+                self._frequency_vector = np.hstack(fvec)
+
+        return self._frequency_vector
+
+    @property
+    def deleteTheseOnModelUpdate(self):
+        toDelete = super(Simulation1DRecursive, self).deleteTheseOnModelUpdate
+        if self.fix_Jmatrix:
+            return toDelete
+
+        if self._Jmatrix is not None:
+            toDelete += ["_Jmatrix"]
+        return toDelete
+
+    # Instantiate
+    def __init__(self, sensitivity_method="2nd order", **kwargs):
+
+        self.sensitivity_method = sensitivity_method
+
+        BaseEMSimulation.__init__(self, **kwargs)
+
+    def _get_recursive_impedances_1d(self, fvec, thicknesses, sigma_1d):
+        """
+        For a given layered Earth model, this returns the complex impedances
+        at the surface for all frequencies.
+
+        :param numpy.ndarray fvec: vector with frequencies in Hz (nFreq,)
+        :param numpy.ndarray thicknesses: layer thicknesses (nLayers-1,)
+        :param numpy.ndarray sigma_1d: layer conductivities (nLayers,)
+        :return numpy.ndarray Z: complex impedances at surface for all frequencies (nFreq,)
+        """
+
+        omega = 2 * np.pi * fvec
+        n_layer = len(sigma_1d)
+
+        # Bottom layer quantities
+        alpha = np.sqrt(1j * omega * mu_0 * sigma_1d[-1])
+        ratio = alpha / sigma_1d[-1]
+        Z = -ratio
```

Comment:
I _think_ this is the opposite convention to what we have currently in the 3D code. (for a halfspace, this will give a +45 degree phase). @grosenkj: do you remember off-hand what the convention is in the 3D code or @JKutt: you might know from your recent simulations / inversions?

Reply:
Yes, I do remember the 3D convention. I made it consistent with SimPEG so `x` is easting and `y` northing and `z` should is positive up. That is like other codes? 
So `Zxy` has a -135 deg phase and `Zyx` has +45deg for a half-space in the 3D code. This is in contrast with the MT world where usually X is North and y is East and Z positive down (for example in EDI files). For the 1D case, since `Zxy=-Zyx` it is neater to have the phase in the 1-quadrant. 
@lheagy does this address your question?

Reply:
Thanks @grosenkj, I think that makes sense. I am looking at developing an implementation that uses boundary conditions to provide the RHS (rather than using the primary-secondary approach), so will just need to do the book-keeping to make sure everything is consistent